gears/gears-kotlin/src/main/kotlin/ApplyIf.kt

```diff
@@ -0,0 +1,54 @@
+package com.redmadrobot.gears.kotlin
+
+/**
+ * Applies the given block of modifications to the T if the condition is true.
+ *
+ * @param condition condition to determine if the block should be applied.
+ * @param block Lambda function that modifies the T.
+ * @return Modified T based on the condition.
+ */
+public inline fun <T> T.applyIf(
+    condition: Boolean,
+    block: T.() -> T,
+): T = applyIfElse(condition = condition, trueBlock = block, falseBlock = { this })
+
+/**
+ * Chooses between two blocks of modifications based on a condition and returns the resulting T.
+ *
+ * @param condition Boolean condition to determine which block to apply.
+ * @param trueBlock Lambda function to modify the T if the condition is true.
+ * @param falseBlock Lambda function to modify the T if the condition is false.
+ * @return Modified T based on the condition.
+ */
+public inline fun <T> T.applyIfElse(
+    condition: Boolean,
+    trueBlock: T.() -> T,
+    falseBlock: T.() -> T,
+): T = if (condition) trueBlock() else falseBlock()
```

Comment:
I'm wondering if it is really worth adding such a function. As it can be simply implemented with `run` + `if`:
```kotlin
modifier.applyIf(addBackground) { background(Color.Blue) }
// vs
modifier.run { if (addBackground) background(Color.Blue) else this }

modifier.applyIfElse(
    condition = addBackground,
    trueBlock = { background(Color.Blue, shape) },
    falseBlock = { clip(shape) },
)
// vs
modifier.run { if (addBackground) background(Color.Blue, shape) else clip(shape) }
```

Version without else is obviously better than `run` + `if`, but `IfElse` version doesn't really make code shorter or clearer in my opinion

Reply:
After discussion, we've decided to remove versions with `else`